utils/zerovec/src/varzerovec/owned.rs

```diff
@@ -324,29 +392,106 @@ mod test {
     use super::VarZeroVecOwned;
     #[test]
     fn test_insert_integrity() {
-        let mut items: Vec<String> = vec![
-            "foo".into(),
-            "bar".into(),
-            "baz".into(),
-            "abcdefghijklmn".into(),
-            "five".into(),
-        ];
-        let mut zerovec = VarZeroVecOwned::from_elements(&items);
-        zerovec.insert(1, &"foo3".into());
-        items.insert(1, "foo3".into());
-        assert_eq!(zerovec, &*items);
+        let mut items: Vec<String> = Vec::new();
+        let mut zerovec = VarZeroVecOwned::new();
 
+        // Insert into an empty vec.
         items.insert(0, "1234567890".into());
         zerovec.insert(0, &"1234567890".into());
         assert_eq!(zerovec, &*items);
 
-        // make sure inserting at the end works
+        zerovec.insert(1, &"foo3".into());
+        items.insert(1, "foo3".into());
+        assert_eq!(zerovec, &*items);
+
+        // Insert at the end.
         items.insert(items.len(), "qwertyuiop".into());
         zerovec.insert(zerovec.len(), &"qwertyuiop".into());
         assert_eq!(zerovec, &*items);
 
         items.insert(0, "asdfghjkl;".into());
         zerovec.insert(0, &"asdfghjkl;".into());
         assert_eq!(zerovec, &*items);
+
+        items.insert(2, "".into());
+        zerovec.insert(2, &"".into());
+        assert_eq!(zerovec, &*items);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_insert_past_end() {
+        VarZeroVecOwned::<String>::new().insert(1, &"".into());
+    }
+
+    #[test]
+    fn test_remove_integrity() {
+        let mut items: Vec<String> = vec![
+            "apples".into(),
+            "bananas".into(),
+            "eeples".into(),
+            "".into(),
+            "baneenees".into(),
+            "five".into(),
+            "".into(),
+        ];
+        let mut zerovec = VarZeroVecOwned::from_elements(&items);
+
+        for index in [0, 2, 4, 0, 1, 1, 0] {
+            items.remove(index);
+            zerovec.remove(index);
+            assert_eq!(zerovec, &*items, "index {}, len {}", index, items.len());
+        }
+    }
+
```

Comment:
suggestion: I'm not entirely sure if the case of insert/remove/replacing at the end is handled -- could you add some tests for that? It can get a little bit tricky there because the last element has no "next element" and instead uses the slice length: which you handle in `element_position_unchecked()` quite well, but I'd like that to be well tested since I definitely had bugs around that in my code.

Reply:
Removing at the end is alreadyhandled (with that third test index of 4 in `test_remove_integrity`). Insert is tested at the end with `qwertyuiop`. Replacing at the end is tested with `0123456789`.

Reply:
Perfect, thanks!